### ISSUE TITLE
Camera conventions & Num objects in the processed GObjaverse

### ISSUE THREAD

**Issue by mengxuyiGit** [autonomousvision/LaRa#8]
Hi Anpei, thanks for the great work and your effort in processing the GOBjaverse!

I have following questions on the processed data:

1. **Camera convention**: the original GObjaverse seems to use a complex left hand system under camera view / world view, do you transform it to another system so that can be directly used for 2DGS rendering?
2. **Normal type**: GOBjaverse provides 2 types of normals: **ng.exr** is the ground truth normal map. on the other hand, **nd.exr** is the normal map adding bump to enhance the detail of geometry surface. Which one is in your processed data?
3. **Dataset size**: the original GObjaverse seems to have 3T, while your processed data is about 1.3T. Do you sample a subset from it?

Thanks!

**Comment by apchenstu** [comment]
Thank you! Regarding the date preprocessing, we use this script to download and convert the format: [prepare_dataset_objaverse](https://github.com/autonomousvision/LaRa/blob/main/tools/prepare_dataset_objaverse.py#L38-L64).

For the size, we only store RGB and normal maps, so it's a little bit smaller than the Gobjaverse.

**Comment by mengxuyiGit** [comment]
Thanks for prompt reply!

I have downloaded the processed h5 file part 1 from the hub, but the loaded c2w are all 0s: https://github.com/autonomousvision/LaRa/blob/c32ae7704faee3423634ba226eefc4dda644f8f5/dataLoader/gobjverse.py#L120-L121 

I have confirmed that the h5 file is not empty and containing 264755 keys, and the idx are valid integers within a reasonable range.

Is there any hint on how to solve this?

**Comment by apchenstu** [comment]
Oh, if you want to download part of the dataset, you have to download the root file "gobjaverse.h5" and the  "gobjaverse_part_xx.h5". Please also refer to gobjaverse.h5 for the dataloader. The gobjaverse.h5 contains camera poses and part_xx contains rendering information.

**Comment by mengxuyiGit** [comment]
Yes, both are present under the ./dataset. I manually changed the range(30) inside the downloading script.

The eval_all.py works fine on my own processed h5 data. But the downloaded part doesn't.

**Comment by mengxuyiGit** [comment]
Is it because the dataloader assumes all the meta information contained in "gobjaverse.h5" are available, which is not true when I have downloaded only partial data?

**Comment by apchenstu** [comment]
> Is it because the dataloader assumes all the meta information contained in "gobjaverse.h5" are available, which is not true when I have downloaded only partial data?

Yes, could be, could you please try the first 10 keys of your download data? they are stored in the first part.

**Comment by apchenstu** [comment]
I just checked there are 8275 samples in part one, please try this scenes_name = np.array(sorted(self.metas.keys()))[:8275] at [here](https://github.com/autonomousvision/LaRa/blob/c32ae7704faee3423634ba226eefc4dda644f8f5/dataLoader/gobjverse.py#L26) .

**Comment by mengxuyiGit** [comment]
I tried the first 1000 and it works. Appreciated!

**Comment by muses0229** [comment]
Hi, I preprocess gobj dataset by my own, and I delete all part file and only retain the merged .h5 file, is it ok for training?

**Comment by apchenstu** [comment]
Yes, all information should already be stored in .h5 after the packing.
